libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1318,11 +1318,39 @@ bool TypeChecker::visit(Conditional const& _conditional)
 	return false;
 }
 
+void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const& _expression)
+{
+	if (auto const* tupleExpression = dynamic_cast<TupleExpression const*>(&_expression))
+	{
+		if (auto const* tupleType = dynamic_cast<TupleType const*>(&_type))
+		{
+			for (size_t i = 0; i < min(tupleExpression->components().size(), tupleType->components().size()); i++)
```

Comment:
Please rather assert that the sizes are the same.

Reply:
I think non-matching sizes are a non-fatal error, so we can't assert that they are the same - but I'll recheck.